sphinx_doc_src/faq.rst

```diff
@@ -112,6 +112,28 @@ You can also declare a local variable in a block::
     end
 
 
+How do I check whether a variable is defined?
+---------------------------------------------
+
+Use ``set -q var``.  For example, ``if set -q var; echo variable defined; end``.  To check multiple variables you can combine with ``and`` and ``or`` like so::
+
+    if set -q var1; or set -q var2
+        echo either variable defined
+    end
+
+Keep in mind that a defined variabled could be empty. Both of these commands define an empty variable: ``set var`` and ``set var ""``.
+
+
+How do I check whether a variable is not empty?
```

Comment:
I see how you're trying to ignore `test`, but I think doing so in this context is a disservice to the readers.

It should at least be mentioned, in something like "for compatibility with other shells, there is also `test -n "$var"`.